src/core/qgsbrowserproxymodel.cpp

```diff
@@ -153,9 +153,28 @@ bool QgsBrowserProxyModel::filterAcceptsRow( int sourceRow, const QModelIndex &s
   if ( !filterAcceptsProviderKey( sourceIndex ) || !filterRootAcceptsProviderKey( sourceIndex ) )
     return false;
 
+  if ( ! mShowLayers )
```

Comment:
Can you add tests for this in TestQgsBrowserProxyModel

Reply:
Yes, I thought about it but the existing test does not use real layer items that I can hide and the test isn't going to be trivial to implement.